File: Data Structures & Algorithms/max-water-container/submission-2.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let l = 0, r = heights.length - 1, output = 0;
        while (l < r) {
            output = Math.max(
                output,
                Math.min(heights[l], heights[r]) * (r - l)
            );

            if (heights[l] < heights[r]) {
                l++;
            } else {
                r--;
            }
        }
        return output;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        l, output, maxF = 0, 0, 0
        currMap = {}

        for r in range(len(s)):
            currMap[s[r]] = currMap.get(s[r], 0) + 1
             
            maxF = max(maxF, currMap[s[r]])

            while (r - l + 1) - maxF > k:
                currMap[s[l]] -= 1
                l += 1
            
            output = max(output, r - l + 1)

        return output